sdk/webpubsub/azure-messaging-webpubsub-client/src/main/java/com/azure/messaging/webpubsub/client/WebPubSubAsyncClient.java

```diff
@@ -0,0 +1,1073 @@
+                         RetryStrategy retryStrategy,
+                         boolean autoReconnect,
+                         boolean autoRestoreGroup) {
+
+        updateLogger(applicationId, null);
+
+        this.applicationId = applicationId;
+
+        this.clientEndpointConfiguration = new ClientEndpointConfiguration(webPubSubProtocol.getName(), userAgent);
+
+        this.clientAccessUrlProvider = Objects.requireNonNull(clientAccessUrlProvider);
+        this.webPubSubProtocol = Objects.requireNonNull(webPubSubProtocol);
+        this.autoReconnect = autoReconnect;
+        this.autoRestoreGroup = autoRestoreGroup;
+
+        this.webSocketClient = webSocketClient == null ? new WebSocketClientNettyImpl() : webSocketClient;
+
+        Objects.requireNonNull(retryStrategy);
+        this.sendMessageRetrySpec = Retry.from(signals -> {
+            AtomicInteger retryCount = new AtomicInteger(0);
+            return signals.concatMap(s -> {
+                Mono<Retry.RetrySignal> ret = Mono.error(s.failure());
+                if (s.failure() instanceof SendMessageFailedException) {
+                    if (((SendMessageFailedException) s.failure()).isTransient()) {
+                        int retryAttempt = retryCount.incrementAndGet();
+                        if (retryAttempt <= retryStrategy.getMaxRetries()) {
+                            ret = Mono.delay(retryStrategy.calculateRetryDelay(retryAttempt))
+                                .then(Mono.just(s));
+                        }
+                    }
+                }
+                return ret;
+            });
+        });
+    }
+
+    /**
+     * Gets the connection ID.
+     *
+     * @return the connection ID.
+     */
+    public String getConnectionId() {
+        return connectionId;
+    }
+
+    /**
+     * Starts the client for connecting to the server.
+     *
+     * @return the task.
+     */
+    public Mono<Void> start() {
+        return this.start(null);
+    }
+
+    Mono<Void> start(Runnable postStartTask) {
+        if (clientState.get() == WebPubSubClientState.CLOSED) {
+            return Mono.error(logger.logExceptionAsError(
+                new IllegalStateException("Failed to start. Client is CLOSED.")));
+        }
+        return Mono.defer(() -> {
+            logger.atInfo()
+                .addKeyValue("currentClientState", clientState.get())
+                .log("Start client called.");
+
+            isStoppedByUser.set(false);
+            isStoppedByUserSink.set(null);
+
+            boolean success = clientState.changeStateOn(WebPubSubClientState.STOPPED, WebPubSubClientState.CONNECTING);
+            if (!success) {
+                return Mono.error(logger.logExceptionAsError(
+                    new IllegalStateException("Failed to start. Client is not STOPPED.")));
+            } else {
+                if (postStartTask != null) {
+                    postStartTask.run();
+                }
+
```

Comment:
Not sure whether it's called in reconnection. SequenceId should reset in reconnection and not rest in recovery

Reply:
Moved to `WebPubSubConnection`, which will be set to null (hence sequenceId cleared) on stop/reconnect (except recover).